### PR TITLE
[Normalizer] Rust kernel and its benchmark against Sklearn

### PR DESCRIPTION
**Pull request by Armin-Ebrahimi-Saba**
Adds Rust l1/l2/max row normalization and a drop-in stratum.Normalizer / stratum.normalize.

- _rust/src/normalize.rs: copy-returning and zero-allocation in-place variants, parallelised over the project thread pool. The l1/l2 row reductions are split into 8 partial sums so LLVM can lower them to NEON, this reassociates the sum, so results differ from a strict left-to-right reduction by a few ULP. Non-contiguous rows take a scalar fallback.
- stratum/adapters/normalizer.py: the Rust path serves dense C-contiguous float32 with axis=1; float64, Fortran order, sparse, axis=0, return_norm and read-only inputs fall back to sklearn with its dtypes and copy semantics preserved.
- stratum/tests/adapters/test_normalizer.py: 40 tests across kernels, normalize(), and the transformer.
- benchmarks/adapters/bench_normalization.py, examples/normalization_example.py.

### CODE REVIEW

File: _rust/src/normalize.rs [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+pub fn normalize_l2(src: ArrayView2<f32>) -> Array2<f32> {
+    let n_rows = src.nrows();
+    let n_cols = src.ncols();
+    let mut out = Array2::<f32>::uninit((n_rows, n_cols));    // uninitialised → no zero-fill pass
+    let pool = get_thread_pool();
+    let mut work = || {
+        out.axis_iter_mut(Axis(0))                           // output rows, mutable (one per task)
+            .into_par_iter()                                 // parallelise across rows
+            .zip(src.axis_iter(Axis(0)))                     // pair each output row with its input row
+            .for_each(|(mut out_row, in_row)| {              // each task: one (out_row, in_row) pair
```
[the remark this answers is not in the document]
Comment (Armin-Ebrahimi-Saba) [reply _rust/src/normalize.rs:92]: This compiles as written — no change needed.

`IndexedParallelIterator::zip` takes `Z: IntoParallelIterator`, not `Z: ParallelIterator`. ndarray (0.16.1, `rayon` feature) provides `impl IntoParallelIterator for AxisIter<'_, A, D>`, so `src.axis_iter(Axis(0))` is converted at the call site; writing `.into_par_iter()` on the RHS would just be doing that conversion explicitly. `Parallel<AxisIterMut>` is `IndexedParallelIterator`, so the pairing is index-aligned, not arbitrary.

Verified on this branch (`acbbcf8`):

- `cargo check --release` — clean
- `cargo test --release` — 6/6 pass, including `l2_copy_matches_inplace`, `l1_unit_norm`, `l2_unit_norm`, `max_unit_norm`

A wrong-zip selection would have to show up as a row/row mismatch, and those tests assert exactly that. The Python suite also checks parity against `sklearn.preprocessing.normalize` for all three norms. Same reasoning covers lines 167 and 202.

File: stratum/__init__.py [hunk 2]
```diff
@@ -26,11 +26,14 @@ def _first_version(*names):
 # Expose our subclasses under the same names
 from .adapters.string_encoder import RustyStringEncoder as StringEncoder
 from .adapters.one_hot_encoder import RustyOneHotEncoder as OneHotEncoder
+from .adapters.normalizer import RustyNormalizer as Normalizer, normalize
```
[the remark this answers is not in the document]
Comment (Armin-Ebrahimi-Saba) [reply stratum/__init__.py:29]: Already done in this PR — README.md is part of the diff:

```diff
    ├─ adapters/             # Public API (dispatch to Rust or fall back to skrub)
    │  ├─ string_encoder.py  # RustyStringEncoder
-   │  └─ one_hot_encoder.py # RustyOneHotEncoder
+   │  ├─ one_hot_encoder.py # RustyOneHotEncoder
+   │  └─ normalizer.py      # RustyNormalizer + normalize()
```

The "Repository Layout" section lists `normalizer.py` alongside the other two adapters. `examples/normalization_example.py` covers the usage side.
